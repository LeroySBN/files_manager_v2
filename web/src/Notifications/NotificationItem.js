import React from "react";
import PropTypes from 'prop-types';
import { StyleSheet, css } from 'aphrodite';


class NotificationItem extends React.PureComponent {
  render() {
    const { type, value, html, markAsRead, id } = this.props;
    return (
      <React.Fragment>
        {type && value ? 
          <li
            className={
              type === "default" ? css(styles.default) : css(styles.urgent)
            }
            onClick={()=> markAsRead(id)}
            data-notification-type={type} >{value}
          </li>
        : null}
        {html ?
          <li
            className={css(styles.urgent)}
            onClick={()=> markAsRead(id)}
            data-notification-type="urgent"
            dangerouslySetInnerHTML={{__html: html}}
          />
        : null}
      </React.Fragment>
    );
  }
}

const styles = StyleSheet.create({
  default: {
    color: 'blue',
    cursor: 'pointer',
    '@media (max-width: 900px)': {
      borderBottom: '1px solid black',
      fontSize: '20px',
      padding: '10px 8px',
    },
  },
  urgent: {
    color: 'red',
    cursor: 'pointer',
    '@media (max-width: 900px)': {
      borderBottom: '1px solid black',
      fontSize: '20px',
      padding: '10px 8px',
    },
  },
});

NotificationItem.propTypes = {
  id: PropTypes.number,
  type: PropTypes.string.isRequired,
  value: PropTypes.string,
  __html: PropTypes.shape({ html: PropTypes.string }),
  markAsRead: PropTypes.func,
};

NotificationItem.defaultProps = {
  type: 'default',
  markAsRead: () => {},
  id: 0,
};

export default NotificationItem;
