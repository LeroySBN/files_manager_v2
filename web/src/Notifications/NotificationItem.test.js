import React from "react";
import { shallow, mount } from "enzyme";
import NotificationItem from "./NotificationItem";
import { StyleSheetTestUtils } from "aphrodite";

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection();
});

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
});


describe("NotificationItem component testing", () => {
  it("NotificationItem renders without crashing", () => {
    const component = shallow(<NotificationItem />);
    expect(component).toMatchSnapshot();
  });

  it('renders the correct html with type and value props', () => {
    const wrapper = shallow(<NotificationItem type="default" value="test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the correct html', () => {
    const wrapper = shallow(<NotificationItem html={{ __html: '<u>test</u>' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("NotificationItem component events testing", () => {  
  it('should call markAsRead with the correct ID when clicked', () => {
    const id = 1;
    const spy = jest.fn();
    const wrapper = mount(
      <NotificationItem
        id={id}
        type="default"
        value="New notification"
        markAsRead={spy}
      />
    );
  
    // Find the list item and simulate a click
    const listItem = wrapper.find('li');
    listItem.simulate('click');
  
    // Verify that the spy was called with the correct ID
    expect(spy).toHaveBeenCalledWith(id);
  });
});
