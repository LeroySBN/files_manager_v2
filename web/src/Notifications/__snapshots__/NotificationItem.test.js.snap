// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationItem component testing NotificationItem renders without crashing 1`] = `ShallowWrapper {}`;

exports[`NotificationItem component testing renders the correct html 1`] = `ShallowWrapper {}`;

exports[`NotificationItem component testing renders the correct html with type and value props 1`] = `ShallowWrapper {}`;

exports[`Notifications component testing NotificationItem renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Notifications component testing renders the correct html 1`] = `ShallowWrapper {}`;

exports[`Notifications component testing renders the correct html with type and value props 1`] = `ShallowWrapper {}`;
